wisp-launchdarkly/src/main/kotlin/wisp/launchdarkly/LaunchDarklyFeatureFlags.kt

```diff
@@ -8,36 +8,50 @@ import com.launchdarkly.sdk.server.interfaces.LDClientInterface
 import com.launchdarkly.shaded.com.google.common.base.Preconditions.checkState
 import com.squareup.moshi.JsonDataException
 import com.squareup.moshi.Moshi
+import io.micrometer.core.instrument.MeterRegistry
+import io.micrometer.core.instrument.Metrics
 import mu.KotlinLogging
 import wisp.feature.*
 import java.util.*
 import java.util.concurrent.Executor
+import kotlin.system.measureTimeMillis
 
 /**
  * Implementation of [FeatureFlags] using LaunchDarkly's Java SDK.
  * See https://docs.launchdarkly.com/docs/java-sdk-reference documentation.
  */
 class LaunchDarklyFeatureFlags constructor(
     private val ldClient: LDClientInterface,
-    private val moshi: Moshi
+    private val moshi: Moshi,
+    private val meterRegistry: MeterRegistry = Metrics.globalRegistry,
 ) : FeatureFlags {
 
     private var featuresWithMigrationWarnings: MutableList<Feature> = mutableListOf()
+    private val launchDarklyClientMetrics = LaunchDarklyClientMetrics(meterRegistry)
 
     fun startUp(): LaunchDarklyFeatureFlags {
-        var attempts = 300
-        val intervalMillis = 100L
-
-        // LaunchDarkly has its own threads for initialization. We just need to keep checking until
-        // it's done. Unfortunately there's no latch or event to wait on.
-        while (!ldClient.isInitialized && attempts > 0) {
-            Thread.sleep(intervalMillis)
-            attempts--
-        }
+        var timedResult = measureTimeMillis {
+            var attempts = 300
+            val intervalMillis = 100L
+
+            // LaunchDarkly has its own threads for initialization. We just need to keep checking until
+            // it's done. Unfortunately there's no latch or event to wait on.
+            while (!ldClient.isInitialized && attempts > 0) {
+                Thread.sleep(intervalMillis)
+                attempts--
+            }
+
+            if (attempts == 0 && !ldClient.isInitialized) {
+                launchDarklyClientMetrics.failedCount.increment()
+                throw Exception("LaunchDarkly did not initialize in 30 seconds")
+            }
 
-        if (attempts == 0 && !ldClient.isInitialized) {
-            throw Exception("LaunchDarkly did not initialize in 30 seconds")
+            // ldClient successfully initialized
+            launchDarklyClientMetrics.successCount.increment()
         }
+        launchDarklyClientMetrics
+            .initializationDuration(meterRegistry)
+            .record(timedResult.toDouble())
```

Comment:
Do you want this to capture the time spent for unsuccessful initializations, the exception above will avoid this?  Not sure if it's desired or not.

Reply:
unsuccessful initializations would always be ~30s and I will just use failure count to capture that.